hazelcast-client/src/test/java/com/hazelcast/client/map/ClientMapBasicTest.java

```diff
@@ -72,765 +82,545 @@ public static void destroy() {
 
     @Test
     public void testClientGetMap() {
-        assertNotNull( client.getMap(randomString()) );
+        assertNotNull(client.getMap(randomString()));
     }
 
     @Test
     public void testGetName() {
-        String mapName = randomString();
-        final IMap map = client.getMap(mapName);
-        assertEquals(mapName, map.getName());
+        assertEquals(randomMapName, randomStringMap.getName());
     }
 
     @Test
     public void testSize_whenEmpty() {
-        final IMap map = client.getMap(randomString());
-        assertEquals(0, map.size());
+        assertEquals(0, randomStringMap.size());
     }
 
     @Test
     public void testSize() {
-        final IMap map = client.getMap(randomString());
-        map.put("key", "val");
-        assertEquals(1, map.size());
+        randomStringMap.put(KEY, NEW_VALUE);
+        assertEquals(1, randomStringMap.size());
     }
 
     @Test
     public void testSize_withMultiKeyPuts() {
-        final IMap map = client.getMap(randomString());
-        final Object key = "Key";
-        final Object oldValue = "oldValue";
-        final Object newValue = "Val";
-
-        map.put(key, oldValue);
-        map.put(key, newValue);
+        randomStringMap.put(KEY, OLD_VALUE);
+        randomStringMap.put(KEY, NEW_VALUE);
 
-        assertEquals(1, map.size());
+        assertEquals(1, randomStringMap.size());
     }
 
     @Test
     public void testIsEmpty_whenEmpty() {
-        final IMap map = client.getMap(randomString());
-        assertTrue(map.isEmpty());
+        assertTrue(randomStringMap.isEmpty());
     }
 
     @Test
     public void testIsEmpty_whenNotEmpty() {
-        final IMap map = client.getMap(randomString());
-        map.put("key", "val");
-        assertFalse(map.isEmpty());
+        randomStringMap.put(KEY, NEW_VALUE);
+        assertFalse(randomStringMap.isEmpty());
     }
 
     @Test
     public void testIsEmpty_afterPutRemove() {
-        final IMap map = client.getMap(randomString());
-        final Object key = "key";
-        map.put(key, "val");
-        map.remove(key);
-        assertTrue(map.isEmpty());
+        randomStringMap.put(KEY, NEW_VALUE);
```

Comment:
Then we cannot move forward anymore. With the mass of test code we already have there will be no way to avoid inconsistency if we start to improve the test quality. But if we don't do this, there will be no progress. The problem is not the inconsistency, but that we have to define how the consistent state should look like in the end (so new tests will be written directly in that way and old ones can be transformed over time).

I think we should discuss how a good test should look like, but **not** in the line notes of this PR.